packages/api/src/learning-materials/adapters/users.adapter.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import { UsersRepository } from '../../users/users.repository';
+
+@Injectable()
+export class UsersAdapter {
```

Comment:
You can just call it Usernames, but UsersAdapter is also OK, but more technical.

Reply:
I haven't noticed that - in Ports & Adapters architecture adapters should implement interface, so rest of the code is dependent on abstraction and Adapter is a glue between what is needed (interface describe it) by module and how is provided.

Reply:
![image](https://docs.google.com/drawings/d/e/2PACX-1vQyv5xAx5hFJPhiK19AGl_2t256M0yKcDSliH8etojltE3tBlEnCndwfsUr1UsXvv5PKGVtrBHkQX3h/pub?w=913&h=129)

https://herbertograca.com/2017/11/16/explicit-architecture-01-ddd-hexagonal-onion-clean-cqrs-how-i-put-it-all-together/